festim/problem.py

```diff
@@ -0,0 +1,164 @@
+import dolfinx
+from dolfinx import fem
+from dolfinx.nls.petsc import NewtonSolver
+import basix
+import ufl
+from mpi4py import MPI
+import numpy as np
+import tqdm.autonotebook
+import festim as F
+
+
+class Problem:
```

Comment:
Should we rename this `ProblemBase` to understand that this is not a class that users should use but only inherit from?

What I mean is that you cannot use `Problem` on its own.

Reply:
yeah, sounds good. I added a short docstring too